### PR TITLE
Enable caching of CPU samples collected at application startup

### LINKED ISSUE #84884: Add `AppStartUp` UserTag to track timing of Flutter app start
Soon, DDS is going to support caching CPU profiles for a given user tag and provide an API for Flutter DevTools to request a cached profile. @bkonyi is working on this. We will use this functionality to support profiling app startup for Flutter apps. In order to use this functionality, we need a [UserTag](https://api.flutter.dev/flutter/dart-developer/UserTag-class.html) `AppStartUp` to be set at the beginning of app start, and unset at the end. This will mark the start and end times for app startup in the CPU profiler. 

I'm not sure where in the flutter ecosystem these UserTag changes will need to take place. If either of them need to happen in Native code in the engine, @bkonyi can make a small change to the C API to support this. @chinmaygarde @iskakaushik

Flutter Tools will then need to pass `AppStartUp`, or whatever the name of the user tag ends up being, as a flag value to DDS, so that DDS knows to cache profiles for that tag. @jonahwilliams

### PR DESCRIPTION
**Pull request by bkonyi**
Fixes https://github.com/flutter/flutter/issues/84884

### CODE REVIEW

**Review by kenzieschmoll — APPROVED** [review]

File: packages/flutter_tools/lib/src/base/dds.dart [hunk 1]
```diff
@@ -52,6 +53,8 @@ class DartDevelopmentService {
           serviceUri: ddsUri,
           enableAuthCodes: disableServiceAuthCodes != true,
           ipv6: ipv6 == true,
+          // Enables caching of CPU samples collected during application startup.
+          cachedUserTags: const <String>['AppStartUp'],
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/base/dds.dart:57]: Is there any documentation of supported tags that we could link to?

Comment (bkonyi) [reply packages/flutter_tools/lib/src/base/dds.dart:57]: In theory, this works for any set `UserTag`, so this is something that could be exposed as a CLI option at some point so users can cache samples associated with any `UserTag` defined in their code. However, I think it's best to just start with this single case so we can make progress on tooling for profiling application startup times.

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/base/dds.dart:57]: Oh, I see, so `UserTag`s are defined in user code?

Comment (bkonyi) [reply packages/flutter_tools/lib/src/base/dds.dart:57]: Yup, but we're setting this particular tag in the engine and resetting to default in the framework after everything's initialized. See [UserTag](https://api.dart.dev/stable/2.13.4/dart-developer/UserTag-class.html).

**Pull request merged**